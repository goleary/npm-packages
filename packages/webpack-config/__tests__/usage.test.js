const {execSync} = require("child_process");
const sh = require("sh-thunk");
const fs = require("fs");

const webpack = __dirname + "/../node_modules/.bin/webpack";

beforeEach(sh({stdio: null})`rm -rf __tests__/fixtures/*/dist/`);

test("can render template", () => {
    execSync(`${webpack} --mode production`, {
        cwd: __dirname + "/fixtures/template",
    });

    const files = fs.readdirSync(__dirname + "/fixtures/template/dist");

    expect(files).toEqual([
        "main.js",
        "main.js.map",
        "manifest.json",
        "out.php",
    ]);

    const out = fs
        .readFileSync(__dirname + "/fixtures/template/dist/out.php")
        .toString();

    expect(out).toMatchSnapshot();
});
