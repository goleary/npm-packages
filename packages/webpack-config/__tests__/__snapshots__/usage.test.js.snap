// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render template 1`] = `
"<?php define( 'JS', 'main.js?v=1aaa704dfce4c5691a01' );
"
`;
